src/backend/distributed/planner/multi_logical_planner.c

```diff
@@ -478,9 +478,10 @@ FullCompositeFieldList(List *compositeFieldList)
 			compositeFieldArray = palloc0(compositeFieldCount * sizeof(bool));
 			relation_close(relation, AccessShareLock);
 
-			for (index = 0; index < compositeFieldCount; index++)
+			for (initFieldIndex = 0; initFieldIndex < compositeFieldCount;
+				 initFieldIndex++)
 			{
-				compositeFieldArray[index] = false;
+				compositeFieldArray[initFieldIndex] = false;
```

Comment:
maybe compositeFieldIndex ?

Reply:
I wanted to do that originally, but that name is already defined in the scope above. I moved that declaration down now to work around that.